<launch>

  <node pkg="odom_visualization" name="odom_visualization" type="odom_visualization" output="screen">
    <remap from="~odom" to="/gazebo_odom" />
    <param name="robot_scale" value="1.0"/>
  </node>
  <!--<node pkg="odom_visualization" name="odom_visualization_plate" type="odom_visualization_plate" output="screen">
    <remap from="~odom" to="manager_node/target_odom" />
    <param name="height" value="1.2"/>
    <param name="width" value="1.6"/>
    <param name="platform_r" value="1.0"/>
  </node>-->
  <include file="$(find odom_visualization)/launch/gazebo_state.launch">
    <!-- <param name="model_name" value="iris"/>
    <param name="model_frame" value="base_link"/>
    <param name="target_frame" value="world"/> -->
  </include>
  
  <rosparam command="load" file="$(find plan_manager)/params/run_hill.yaml" />
  <rosparam command="load" file="$(find plan_manager)/params/occ_map.yaml" />

  <arg name="global_env_pcd2_topic" value="/origin_map" />


  <node pkg="plan_manager" type="manager_node" name="manager_node" output="screen">
    <remap from="~odom" to="odom"/>
    <remap from="~heartbeat" to="heartbeat"/>
    <remap from="~trajectory" to="trajectory"/>
    <remap from="~replanState" to="replanState"/>
    <remap from="~triger" to="/triger"/>
    <remap from="~land_triger" to="/land_triger"/>
    <remap from="~target" to="/target/odom"/>
    <remap from="/global_cloud" to="/origin_map"/>

    <param name="uneven_map/map_pcd" type="string" value="$(find uneven_map)/maps/collision.pcd" />
    <param name="uneven_map/map_file" type="string" value="$(find uneven_map)/maps/collision.map" />

    <param name="plan_hz" value="5"/>

    <param name="K" value="16"/>
    <param name="vmax" value="3.0"/>
    <param name="amax" value="5.0"/>

    <param name="thrust_max" value="17.0"/>
    <param name="thrust_min" value="5.0"/>
    <param name="omega_max" value="3.0"/>
    <param name="omega_yaw_max" value="0.5"/>

    <param name="robot_l" value="0.02"/>
    <param name="robot_r" value="0.15"/>
    <param name="platform_r" value="1.0"/>
    <param name="v_plus" value="0.1"/>

    <param name="rhoT" value="100000.0"/>
    <param name="rhoP" value="10000000.0"/>
    <param name="rhoV" value="1000.0"/>
    <param name="rhoA" value="1000.0"/>
    <param name="rhoVt" value="1000000.0"/>
    <param name="rhoThrust" value="10000.0"/>
    <param name="rhoOmega" value="100000.0"/>
    <param name="rhoPerchingCollision" value="1000000.0"/>

    <param name="replan" value="false"/>
    <param name="pause_debug" value="false"/>
    
    <remap from="/global_cloud" to="$(arg global_env_pcd2_topic)"/> 
  </node>
  <!-- trajectory server -->
  <node pkg="plan_manager" name="traj_server" type="traj_server" output="screen">
      <param name="time_forward" value="0.1" type="double"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="map_to_world_transform"
      args="0 0 0 0 0 0 /map /world 10" />
  
</launch>